### ISSUE TITLE
Accessibility improvements on Pagination component

### ISSUE THREAD

**Issue by muratcorlu** [Trendyol/baklava#293]
- [ ] Pagination component should be usable with keyboard.
- [ ] Pagination component should use related `aria-` attributes.

Some helpful information: https://www.magentaa11y.com/checklist-web/pagination/
